### PR TITLE
[SYCL][Driver] Update directory name to integration header

### PR DESCRIPTION
**Pull request by mdtoguchi**
There is a possibility that the TMP environment variable on Windows when set
does not exactly match the real path representation of the directory via
case sensitivity.  Fixup the directory/filename to match real path expectations
to reduce the potential warnings emitted during compilation.

> clang++ -fsycl -c accessors.cpp
<built-in>:1:10: warning: non-portable path to file
      '"c\\Users\\AppData\Local\Temp\3\accessors-62b196.h"'; specified
      path differs in case from file name on disk [-Wnonportable-include-path]
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "c\\Users\AppData\Local\Temp\3\accessors-62b196.h"

Signed-off-by: Michael D Toguchi <michael.d.toguchi@intel.com>

### CODE REVIEW

File: clang/test/Driver/sycl-offload-header-check.cpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: system-windows
+// Test the integrated header file name generation.  Name should match the
+// actual path name and not the environment variable setting
+// RUN: mkdir -p %t_DiRnAmE
+// invoke the compiler overriding output temp location
+// RUN: env TMPDIR=%t_dirname TEMP=%t_dirname TMP=%t_dirname  \
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload-header-check.cpp]: To clarify: are all of these ENVs needed?

File: clang/test/Driver/sycl-offload-header-check.cpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+// RUN: %clang_cl -### -fsycl %s 2>&1 | \
+// RUN: FileCheck --check-prefix=CHECK-HEADER %s
+// CHECK-HEADER: clang{{.*}} "-fsycl-int-header=[[HEADER:.+\.h]]"
+// CHECK-HEADER-NOT: clang{{.*}} "-include" "[[HEADER]]"
+// CHECK-HEADER: clang{{.*}} "-include" "{{.*}}_DiRnAmE{{.+}}.h"
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload-header-check.cpp]: I may be missing something, but is the following implied?
```suggestion
// CHECK-HEADER: clang{{.*}} "-include" "{{.*}}_DiRnAmE\[[HEADER]]"
```
If this is the intended behaviour, than I'd suggest going with this CHECK to add more precision

File: clang/test/Driver/sycl-offload-header-check.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: system-windows
+// Test the integrated header file name generation.  Name should match the
+// actual path name and not the environment variable setting
+// RUN: mkdir -p %t_DiRnAmE
+// invoke the compiler overriding output temp location
+// RUN: env TMPDIR=%t_dirname TEMP=%t_dirname TMP=%t_dirname  \
+// RUN: %clang_cl -### -fsycl %s 2>&1 | \
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload-header-check.cpp]: Also to clarify: is `%clang` never invoked on Windows?

File: clang/test/Driver/sycl-offload-header-check.cpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: system-windows
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload-header-check.cpp]: Is Linux unaffected/guarded by the existing LITs?

File: clang/test/Driver/sycl-offload-header-check.cpp [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+// Test the integrated header file name generation.  Name should match the
```
Comment (Fznamznon) [inline clang/test/Driver/sycl-offload-header-check.cpp]:
```suggestion
// Test the integration header file name generation.  Name should match the
```

Comment (AGindinson) [reply clang/test/Driver/sycl-offload-header-check.cpp]: +1. I think amending the commit message might be necessary as well.

File: clang/test/Driver/sycl-offload-header-check.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-header-check.cpp]: I believe only TMP is needed - I will clean up.

File: clang/test/Driver/sycl-offload-header-check.cpp (hunk 2 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-header-check.cpp]: The contents of [[HEADER]] is the full path plus header name.  I just want to verify that the name being passed is the actual path name and not the one as set by the environment variable (_dirname)

File: clang/test/Driver/sycl-offload-header-check.cpp (hunk 3 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-header-check.cpp]: You can run %clang on Windows.   I will add a test for it.

File: clang/test/Driver/sycl-offload-header-check.cpp (hunk 4 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-header-check.cpp]: Test is not valid on Linux.  The compiler will complain based the TMP env variable pointing to an invalid directory (case mismatch)

Comment (AGindinson) [reply clang/test/Driver/sycl-offload-header-check.cpp]: Ah, I rather meant "is Linux sure to be unaffected by the code changes"

Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-header-check.cpp]: Thanks for clarifying.  The code path in question is for Linux as well, but should be well tested with existing integration header checks in sycl-offload.c

**Pull request merged**